go-controller/pkg/cni/cni.go

```diff
@@ -136,8 +136,11 @@ func (pr *PodRequest) cmdAdd(kubeAuth *KubeAPIAuth, podLister corev1listers.PodL
 	if err != nil {
 		return nil, fmt.Errorf("failed to get pod annotation: %v", err)
 	}
-	if err := pr.checkOrUpdatePodUID(podUID); err != nil {
-		return nil, err
+	pod, err := getPod(podLister, kclient, pr.PodNamespace, pr.PodName)
+	if err == nil && !IsStaticPod(pod) {
```

Comment:
coming back to the main logic here, was the decision to totally skip the check for static pods or was it to use the mirror hash: kubernetes.io/config.mirror: a456b77dd3038db3f5590ca665f2e3b4 versus the uid in the cni request: 
```
"status": {
    "id": "196c503c05f592ac97a454732b2839e914eb1a85da6cc270009b31a72d008557",
    "metadata": {
      "attempt": 0,
      "name": "kube-apiserver-ovn-control-plane",
      "namespace": "kube-system",
      "uid": "a456b77dd3038db3f5590ca665f2e3b4"
    },
```
instead of the uid checks we do for normal pods.

```
hash := getPodHash(pod)
	copyPod.Annotations[kubetypes.ConfigMirrorAnnotationKey] = hash

func getPodHash(pod *v1.Pod) string {
	// The annotation exists for all static pods.
	return pod.Annotations[kubetypes.ConfigHashAnnotationKey]
}
```

Reply:
we need to skip the check for static pods, as well as use the uid that master is expecting from the node side